data/expense/src/main/java/com/kappzzang/jeongsan/repositoryimpl/ExpenseListRepositoryImpl.kt

```diff
@@ -26,41 +24,55 @@ class ExpenseListRepositoryImpl @Inject constructor(
     private fun getJwt(): String = auth.getSavedJwt()
     private fun isJwtValid(jwt: String): Boolean = (jwt != "")
 
-    private fun mapResponseBody(body: ExpenseListResponseDTO): ExpenseListResponse {
-        val expenses = body.expenseList.map { ExpenseEntityMapper.mapExpenseEntityToModel(it) }
-        return ExpenseListResponse(
-            totalExpenseToSend = body.totalPrice.toInt(),
-            expenseList = expenses,
-            totalPrice = body.totalPrice.toInt()
-        )
+    override suspend fun uploadExpense(receiptItem: ReceiptItem, groupId: String): String {
+        val jwt = getJwt()
+        if (!isJwtValid(jwt)) {
+            return ""
+        }
+        val response = dataSource.addExpense(receiptItem, jwt, groupId)
+        when (response.code()) {
+            201 -> response.body()?.expenseId?.let {
+                return it
+            } ?: throw IllegalStateException("알 수 없는 오류 발생: ${response.message()}")
+
+            400 -> throw IllegalArgumentException("유효하지 않는 입력 값")
+            404 -> throw IllegalStateException(response.message())
+            500 -> throw IllegalStateException(response.message())
+            else -> {
+                if (response.code() / 100 == 2) {
+                    response.body()?.expenseId?.let {
+                        return it
+                    } ?: throw IllegalStateException("알 수 없는 오류 발생: ${response.message()}")
+                } else {
+                    throw IllegalStateException("알 수 없는 오류 발생: ${response.message()}")
+                }
+            }
+        }
     }
```

Comment:
P2: 업로드 성공 실패 여부를 UseCase에서 전달하려면 값을 반환해주어야할 것 같아요~ 추가적으로 201 일때와 response.code() /100 == 2 가 수행할 동작이 동일하다면 아래처럼 고쳐도 될 것 같아요
```suggestion
override suspend fun uploadExpense(receiptItem: ReceiptItem, groupId: String): String? {
    return try {
        val jwt = getJwt()
        if (!isJwtValid(jwt)) {
            return null
        }

        val response = dataSource.addExpense(receiptItem, jwt, groupId)
        return when (response.code()) {
            in 200..299 -> response.body()?.expenseId?.let {
                return it
            } ?: throw IllegalStateException("알 수 없는 오류 발생: ${response.message()}")
            400 -> throw IllegalArgumentException("유효하지 않는 입력 값")
            404 -> throw IllegalStateException(response.message())
            500 -> throw IllegalStateException(response.message())
            else -> throw IllegalStateException("알 수 없는 오류 발생: ${response.message()}")
        }
    } catch (e: Exception) {
        e.printStackTrace()
        null
    }
}
```

Reply:
실패시 이유에 따라 처리할 경우가 있을 수 있겠네요! 거기까지는 생각을 못해봤네요. 좋은 의견 감사합니다!ㅎㅎ 바꾼다면 말씀하신대로 수현님 방식대로 하는 것이 좋을 것 같습니다!

Reply:
좋은 의견 감사합니다. 네트워크 에러에 대한 처리가 확실히 미흡한 것 같네요.
저의 경우엔, UseCase에서 try-catch 문을 이용해 에러 처리를 수행하는 방식을 생각했습니다. 
하지만, 수현님의 방식처럼 데이터 레이어 내부에서 에러 핸들링을 끝내고, 성공/실패 여부를 포함한 Result 객체를 리턴하는 방식이 더 적절한 것 같아요.
단순히 실패 시 null을 리턴하여도 좋지만, 나중에 실패 이유에 따라 뷰모델에서 유동적으로 처리를 해야할 수도 있으니 (ex: 인증 오류면 다시 로그인, 네트워크 오류면 재시도, 404면 취소) Result 객체에 실패 이유를 담아서 리턴하는 방식이 좋지 않을까 생각합니다.